src/sweetalert2.js

```diff
@@ -597,6 +597,20 @@ const sweetAlert = (...args) => {
       } else {
         error('Unexpected type of inputOptions! Expected object, Map or Promise, got ' + typeof params.inputOptions)
       }
+    } else if (['text', 'email', 'number', 'tel', 'textarea'].includes(params.input) && params.inputValue instanceof Promise) {
+      sweetAlert.showLoading()
+      dom.hide(input)
+      params.inputValue.then((inputValue) => {
+        input.value = inputValue + ''
```

Comment:
I think we should handle numbers correctly if input type is `'number'`:

```js
input.value = params.input === 'number' ? parseFloat(inputValue) : inputValue + ''
```

Reply:
Very good point. Should i make it more robust handling the case `parseFloat()` doesn't return a number: 
```
input.value = params.input === 'number' ? parseFloat(inputValue) || 0 : inputValue + ''
```

Reply:
Good catch, yes please.